deepmd/model/ener.py

```diff
@@ -268,6 +281,114 @@ def build(
             )
             force = force + sw_force + tab_force
 
+        force = tf.reshape(force, [-1, 3 * natoms[1]])
+        if self.spin is not None:
```

Comment:
Please do not write everything in the body of `build`. Pack all the code in the condition to a new method, and provide a clear interface letting people know what does the method do.